Sources/MapboxCoreNavigation/RouteController.swift

```diff
@@ -668,6 +668,30 @@ extension RouteController: Router {
                 self.isRerouting = false
             }
         }
+        
+        switch rerouteRequest {
+        case .default:
+            calculateRoutes(from: location, along: progress, completion: calculationCompletion)
+        case .custom(let reroutingResult):
+            DispatchQueue.global().async {
+                let (options, result) = reroutingResult()
+                let session = Directions.Session(options: options,
+                                                 credentials: NavigationSettings.shared.directions.credentials)
+                
+                DispatchQueue.main.async {
+                    switch result {
+                    case .failure(let error):
+                        return calculationCompletion(session, .failure(error))
+                    case .success(let response):
+                        guard let mostSimilarIndex = response.routes?.index(mostSimilarTo: progress.route) else {
```

Comment:
NCS pointed out that this algorithm can sometimes select the route with the biggest ETA, should we somehow allow them to select one of the routes using their own algorithm?
Currently, it's a little bit misleading that we call `willReroute` callback not before the reroute itself, but before we start to fetch possible reroutes.
Maybe we should rename `willReroute` to `willFetchReroutes` and add a new `willReroute` callback where an application can select the best route.🤔

Reply:
A bit ugly, but user can achieve this if they just remove "slower" routes from `reroutingResult`.

We could (and actually will) revisit rerouting logic once we incorporate NavNative's rerouting implementation.